src/chars.rs

```diff
@@ -128,15 +128,29 @@ pub(crate) fn count_impl<T: ByteChunk>(text: &[u8]) -> usize {
             inv_count += ((byte & 0xC0) == 0x80) as usize;
         }
 
-        // Take care of the middle bytes in big chunks.
-        for chunks in middle.chunks(T::MAX_ACC) {
-            let mut acc = T::zero();
-            for chunk in chunks.iter() {
-                acc = acc.add(chunk.bitand(T::splat(0xc0)).cmp_eq_byte(0x80));
+        let char_boundaries = |val: &T| val.bitand(T::splat(0xc0)).cmp_eq_byte(0x80);
+
+        // Take care of the middle bytes in big chunks. Loop unrolled
+        for chunks in middle.chunks_exact(4) {
+            let mut iter = chunks.iter();
+            while let Some(chunk) = iter.next() {
+                let val1 = char_boundaries(chunk);
+                let val2 = char_boundaries(iter.next().unwrap());
+                let val3 = char_boundaries(iter.next().unwrap());
+                let val4 = char_boundaries(iter.next().unwrap());
+                let val1_2 = val1.add(val2);
+                let val3_4 = val3.add(val4);
+                inv_count += val1_2.add(val3_4).sum_bytes();
             }
```

Comment:
I might be missing something, but I don't think this secondary iterator and inner loop are needed.  I think this should be sufficient:

```rust
let val1 = char_boundaries(chunks[0]);
let val2 = char_boundaries(chunks[1]);
let val3 = char_boundaries(chunks[2]);
let val4 = char_boundaries(chunks[3]);
let val1_2 = val1.add(val2);
let val3_4 = val3.add(val4);
inv_count += val1_2.add(val3_4).sum_bytes();
```

Further, the `val1_2` temporaries don't IMO add much readability.  So maybe just this:

```rust
let val1 = char_boundaries(chunks[0]);
let val2 = char_boundaries(chunks[1]);
let val3 = char_boundaries(chunks[2]);
let val4 = char_boundaries(chunks[3]);
inv_count += val1.add(val2).add(val3.add(val4)).sum_bytes();
```